integration-tests/src/test/java/org/openmrs/module/fhir2/providers/r4/ServiceRequestFhirResourceProviderIntegrationTest.java

```diff
@@ -216,4 +221,66 @@ public void shouldReturnCountForServiceRequestAsXml() throws Exception {
 		assertThat(result.getType(), equalTo(Bundle.BundleType.SEARCHSET));
 		assertThat(result, hasProperty("total", equalTo(4)));
 	}
+	
+	@Test
+	public void shouldReturnAnEtagHeaderWhenRetrievingAnExistingServiceRequest() throws Exception {
+		MockHttpServletResponse response = get("/ServiceRequest/" + SERVICE_REQUEST_UUID).accept(FhirMediaTypes.JSON).go();
+		
+		assertThat(response, isOk());
+		assertThat(response.getContentType(), is(FhirMediaTypes.JSON.toString()));
+		
+		assertThat(response.getHeader("etag"), notNullValue());
+		assertThat(response.getHeader("etag"), startsWith("W/"));
+		
+		assertThat(response.getContentAsString(), notNullValue());
+		
+		ServiceRequest serviceRequest = readResponse(response);
+		
+		assertThat(serviceRequest, notNullValue());
+		assertThat(serviceRequest.getMeta().getVersionId(), notNullValue());
+		assertThat(serviceRequest, validResource());
+	}
+	
+	@Test
+	public void shouldReturnNotModifiedWhenRetrievingAnExistingServiceRequestWithAnEtag() throws Exception {
+		MockHttpServletResponse response = get("/ServiceRequest/" + SERVICE_REQUEST_UUID).accept(FhirMediaTypes.JSON).go();
+		
+		assertThat(response, isOk());
+		assertThat(response.getContentType(), is(FhirMediaTypes.JSON.toString()));
+		assertThat(response.getContentAsString(), notNullValue());
+		assertThat(response.getHeader("etag"), notNullValue());
+		
+		String etagValue = response.getHeader("etag");
+		
+		response = get("/ServiceRequest/" + SERVICE_REQUEST_UUID).accept(FhirMediaTypes.JSON).ifNoneMatchHeader(etagValue).go();
+		
+		assertThat(response, isOk());
+		assertThat(response, statusEquals(HttpStatus.NOT_MODIFIED));
+	}
+	
+	@Ignore
```

Comment:
not sure if this is allowed here, but it is because `ServiceRequest` [doesn't support update yet](https://github.com/openmrs/openmrs-module-fhir2/blob/7acc81d842689f498700eb562e928fd03c82ef99/api/src/main/java/org/openmrs/module/fhir2/providers/r4/ServiceRequestFhirResourceProvider.java#L77-L85)